### PR TITLE
Fix -Dbuild=library target

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
Can we please add a ci target for library to make sure it doesn't break again?

**Comment by hughsie** [comment]
> add a ci target

As in one more into the matrix? Or edit the Fedora one to re-do the build with `meson configure -Dbuild=library` perhaps?

**Comment by hughsie** [comment]
> I think a new target is probably fine for this.

Actually, if we use the existing fedora container we can do a 23-target initial build as a smoke-test before doing all the other stuff -- that would quickly fail in the case we forgot to add a source file or something drastic.

File: contrib/ci/fedora.sh [hunk 1]
```diff
@@ -16,7 +16,16 @@ if [ "$QUBES" = "true" ]; then
     QUBES_MACRO=(--define "qubes_packages 1")
 fi
 
+# smoke test something small with no auto-deps
 meson setup .. \
+    -Dbuild=library \
+    -Dauto_features=disabled \
+    -Dtests=false $@
```
Comment (superm1) [inline contrib/ci/fedora.sh:23]: do we actually need to pass arguments to this?

**Review by superm1 — DISMISSED** [review]
one nit, otherwise LGTM

File: contrib/ci/fedora.sh (hunk 1 above)
Comment (hughsie) [reply contrib/ci/fedora.sh]: Nope; removed, thanks.

**Pull request merged**